From 759b78278fa15ad6460069b4c3139034a7ef460e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:34 -0400
Subject: [tools] perf: Overload pr_stat traceevent print function

Message-id: <1408953210-26343-87-git-send-email-jolsa@redhat.com>
Patchwork-id: 88787
O-Subject: [PATCH RHEL7.1 BZ1133083 086/322] perf tools: Overload pr_stat traceevent print function
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f772abc63462a48df605259fbe1af3fe9b223926
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Dec 3 14:09:25 2013 +0100

description
===========
The traceevent lib uses pr_stat to display all standard info. It's
defined as __weak. Overloading it with perf version plugged into perf
output system logic.

Displaying the pr_stat stuff under '-v' option.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/debug.c b/tools/perf/util/debug.c
index 399e74c..8640a91 100644
--- a/tools/perf/util/debug.c
+++ b/tools/perf/util/debug.c
@@ -16,13 +16,11 @@
 int verbose;
 bool dump_trace = false, quiet = false;
 
-int eprintf(int level, const char *fmt, ...)
+static int _eprintf(int level, const char *fmt, va_list args)
 {
- va_list args;
  int ret = 0;
 
  if (verbose >= level) {
-  va_start(args, fmt);
   if (use_browser >= 1)
    ui_helpline__vshow(fmt, args);
   else
@@ -33,6 +31,32 @@ int eprintf(int level, const char *fmt, ...)
  return ret;
 }
 
+int eprintf(int level, const char *fmt, ...)
+{
+ va_list args;
+ int ret;
+
+ va_start(args, fmt);
+ ret = _eprintf(level, fmt, args);
+ va_end(args);
+
+ return ret;
+}
+
+/*
+ * Overloading libtraceevent standard info print
+ * function, display with -v in perf.
+ */
+void pr_stat(const char *fmt, ...)
+{
+ va_list args;
+
+ va_start(args, fmt);
+ _eprintf(1, fmt, args);
+ va_end(args);
+ eprintf(1, "\n");
+}
+
 int dump_printf(const char *fmt, ...)
 {
  va_list args;
diff --git a/tools/perf/util/debug.h b/tools/perf/util/debug.h
index efbd988..443694c 100644
--- a/tools/perf/util/debug.h
+++ b/tools/perf/util/debug.h
@@ -17,4 +17,6 @@ void trace_event(union perf_event *event);
 int ui__error(const char *format, ...) __attribute__((format(printf, 1, 2)));
 int ui__warning(const char *format, ...) __attribute__((format(printf, 1, 2)));
 
+void pr_stat(const char *fmt, ...);
+
 #endif /* __PERF_DEBUG_H */
-- 
1.7.1